src/System.Buffers/src/System/Buffers/DefaultArrayPoolBucket.cs

```diff
@@ -80,7 +96,17 @@ internal void Return(T[] buffer)
                 // we can just drop this buffer
                 if (_index != 0)
                 {
+                    // If we have an event listener, we are in "debug" mode, so do a safety check
```

Comment:
Hmm that's a very good point, I'll switch to using the Debugger.IsAttached property. Do we have a way of mocking that property so I can test that the double return check it guards works in a Unit Test?